sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp

```diff
@@ -438,8 +450,9 @@ __esimd_scatter_scaled(__SEIEED::simd_mask_storage_t<N> pred,
     char *SlmBase = I->__cm_emu_get_slm_ptr();
     for (int i = 0; i < N; ++i) {
       if (pred[i]) {
-        Ty *addr = reinterpret_cast<Ty *>(elem_offsets[i] + SlmBase);
-        *addr = vals[i];
+        ActualTy *addr =
```

Comment:
here and in other places:
```suggestion
        OrigTyAsInt *addr =
```

Note that what you call `ActualTy` is not actual type, this information is lost. You only know the size in bytes of the original type.